### DISCUSSION TITLE
V7 - Controller: Validation doesn't update when using onChange and value

### DISCUSSION THREAD

**Discussion by knoefel** [react-hook-form/react-hook-form#4574] [Q&A]
I noticed the following and i'm not sure if i'm doing something wrong or it is a bug:

When I use `onChange` and `value` of `field` inside the `render` prop function of the `Controller` component, validation is not executed after I fix a validation error in another field. It is easier to understand with an example:

Steps to reproduce:

1. open [CodeSandbox](https://codesandbox.io/s/react-hook-form--validation-bug-v7-mzfn8?file=/src/index.js)
2. remove the value from the first input with the label 'Name'.
3. `formState.isValid` will update to `false`
4. type something in the input again
5. `formState.isValid` doesn't update to `true`

This only happens when I use `onChange` and `value`, if I just spread the `field` object into the `inputProps` prop (as in my case i use Material UI) the validation works and the formState updates itself after input changes.

**Comment by bluebill1049** [comment]
field props should assigned to `onChange`, `value` instead innerRef

**Reply by knoefel** [reply]
I'm not sure if i understand you correctly, `onChange` and `value` are assigned to the component (for the 2 date fields). When i change the `Textfield` to also use `onChange` and `value` nothing changes. It still doesn't update the `formState.isValid` after clearing the form error. Here is the updated [CodeSandbox](https://codesandbox.io/s/react-hook-form--validation-bug-v7-forked-4v7nq?file=/src/index.js).

**Reply by bluebill1049** [reply]
<img width="1006" alt="Screen Shot 2021-04-01 at 5 41 50 pm" src="https://user-images.githubusercontent.com/10513364/113253822-8cbf3380-9311-11eb-934a-52fcaff9d0f1.png">

looks like it's working for me now.

**Reply by knoefel** [reply]
No, unfortunately not. `formState.isValid` will stay `false` if you enter some characters in the Name `TextField` again. It only works the first time when it changes from `true` to `false`.

**Reply by kai23** [reply]
Hey @bluebill1049 

I think I get the same "error" here : https://codesandbox.io/s/react-hook-form-v7-controller-forked-2n3id?file=/src/index.js

As you can see, the validation just never update. But if you submit the form, the datas are correctly set.
I've tried to change the `mode` but with the `onChange` mode, it's even weirder : the form is valid, although the data object is "null" with a `required` rule

**Reply by bluebill1049** [reply]
@kai23 i think you are missing `defaultValues` at `useForm`.

**Reply by kai23** [reply]
sadly, same thing :  https://codesandbox.io/s/react-hook-form-v7-controller-forked-zsslb?file=/src/index.js
I swear I'm trying to figure out why, but can't get it working :/

**Reply by bluebill1049** [reply]
@kai23 https://codesandbox.io/s/react-hook-form-v7-controller-forked-kmxk8?file=/src/index.js you are missing mode `onChange` 👍

**Reply by bluebill1049** [reply]
@knoefel were your issue above ever get resolved? sorry, I got so many notifications each day, I probably have missed it...

**Reply by knoefel** [reply]
@bluebill1049 No need to apologize. As always, you resolved it instantly with this bugfix: #4582 👍

**Reply by bluebill1049** [reply]
oh cool! thanks @knoefel for the confirmation.

**Reply by kai23** [reply]
It's also alright to me ! You're the best, many thanks.
